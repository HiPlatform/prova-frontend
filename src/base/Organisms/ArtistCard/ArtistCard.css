.artistCard {
  display: flex;
  width: 310px;
  height: 430px;
  flex-direction: column;
  margin: 10px 4px;
  align-items: stretch;
  background-color: #f5f5f5;
  border-bottom: 1px solid #ddd;
  border-radius: 5px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.2);
  flex-grow: 1;
  max-width: 620px; }
  .artistCard > img {
    width: 100%;
    height: 280px; }
  .artistCard__info {
    width: 100%;
    height: 100%;
    position: relative;
    padding: 6px; }
    .artistCard__info__title > .attributeField, .artistCard__info__extra > .attributeField {
      margin: 4px 0;
      padding-right: 4px;
      flex: 1 auto; }
    .artistCard__info__title {
      font-size: 13px;
      font-weight: bold; }
    .artistCard__info__extra {
      font-size: 11px; }
    .artistCard__info__actions {
      width: 100%;
      position: absolute;
      bottom: 6px; }
