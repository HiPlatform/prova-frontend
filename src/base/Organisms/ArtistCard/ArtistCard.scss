.artistCard {
  display: flex;
  width: 310px;
  height: 430px;
  flex-direction: column;
  margin: 10px 4px;
  align-items: stretch;
  border: 1px solid #d9dadb;
  border-radius: 5px;
  box-shadow: 0px 1px 0px 0px #d9dadb;
  flex-grow: 1;
  max-width: 620px;

  & > img {
    width: 100%;
    height: 280px;
  }

  &__info {
    width: 100%;
    height: 100%;
    position: relative;
    padding: 6px;

    &__title > .attributeField, &__extra > .attributeField {
      margin: 4px 0;
      padding-right: 4px;
      flex: 1 auto;
    }

    &__title {
      font-size: 13px;
      font-weight: bold;
    }

    &__extra {
      font-size: 11px;
    }

    &__actions {
      width: 100%;
      position: absolute;
      bottom: 6px;
    }
  }
}