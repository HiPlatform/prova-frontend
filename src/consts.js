export const URLS = {
  SEARCH: 'https://api.spotify.com/v1/search',
};

export const CREDENTIALS = {
  headers: {
    Authorization: 'Bearer BQDGGAag6Hocue0iXDFWp0O9CdmjCSq6nI9R2-HE-_N-skmHG8V3btyKOZnilxWjn7cpkFMAbM70G8CH_-Y',
  },
};

export const HOMEPAGE_ACTIONS = {
  SEARCH_DATA: 'SEARCH_DATA',
  SEARCH_TYPE: 'SEARCH_TYPE',
  SEARCH_CHANGE: 'SEARCH_CHANGE',
  SEARCH_DETAILS: 'SEARCH_DETAILS',
  NAVIGATE: 'NAVIGATE',
};
