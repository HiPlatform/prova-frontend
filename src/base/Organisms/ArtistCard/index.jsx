import React from 'react';
import PropTypes from 'prop-types';

import { checkPopularity, imageUrl } from '../../../helpers/cardsFunctions';

import Button from '../../Atoms/Button';
import Image from '../../Atoms/Image';
import AttributeField from '../../Molecules/AttributeField';

import './ArtistCard.css';
import FavoriteButton from '../../Molecules/FavoriteButton';

const ArtistCard = ({
  name,
  popularity,
  images,
  genres,
  href,
  details,
  item,
}) => (
  <div className="artistCard">
    <FavoriteButton item={item} list="artist" />
    <Image url={imageUrl(images)} alt="album" />
    <div className="artistCard__info">
      <div className="artistCard__info__title">
        <AttributeField text={name} />
      </div>
      <div className="artistCard__info__extra">
        <AttributeField label="Genres" text={genres.join(', ')} />
        <AttributeField label="Popularity" text={checkPopularity(popularity)} />
      </div>
      <div className="artistCard__info__actions">
        <Button onClick={() => details(`${href}/albums?limit=5`)}>MORE DETAILS</Button>
      </div>
    </div>
  </div>
);

ArtistCard.propTypes = {
  name: PropTypes.string.isRequired,
  popularity: PropTypes.number.isRequired,
  images: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
  })).isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  href: PropTypes.string.isRequired,
  details: PropTypes.func.isRequired,
  item: PropTypes.shape().isRequired,
};

export default ArtistCard;
