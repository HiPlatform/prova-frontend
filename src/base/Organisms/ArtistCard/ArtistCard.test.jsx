import React from 'react';
import { shallow } from 'enzyme';
import ArtistCard from './index';
import { DEFAULT_TEXTS, POPULARITY_RANGES } from '../../../consts';

describe('ArtistCard', () => {
  const details = jest.fn();
  const props = {
    name: 'John Due',
    popularity: 90,
    images: [{ url: 'https://some.image/url' }],
    genres: ['foo', 'bar', 'fuzz'],
    href: 'https://some.redirect/url',
    details,
  };
  const component = shallow(<ArtistCard {...props} />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the details function when the "more details" button is clicked', () => {
    component.find('Button').simulate('click');
    expect(details).toHaveBeenCalled();
  });

  it('should put the first image url in card header', () => {
    expect(component.find('Image').dive().prop('src')).toBe(props.images[0].url);
  });

  it('should put a mock image url in card header if prop images has no images at all', () => {
    const images = [];
    component.setProps({ images });

    expect(component.find('Image').dive().prop('src')).toBe(DEFAULT_TEXTS.IMAGE);
  });

  it('should render the artist name given in the props', () => {
    expect(component.find('.artistCard__info__title').render().text()).toBe(props.name);
  });

  it('should render all the genres', () => {
    const genreElement = component.find('.artistCard__info__extra')
      .find({ label: 'Genres' })
      .render()
      .find('p');
    expect(genreElement.text()).toBe(props.genres.join(', '));
  });

  it('should render nothing when the genres prop is an empty array', () => {
    const genres = [];
    component.setProps({ genres });
    const genreElement = component.find('.artistCard__info__extra')
      .find({ label: 'Genres' })
      .render()
      .find('p');

    expect(genreElement.text()).toBe('');
  });

  it('should render "Hot" when the popularity rate is above 80', () => {
    const popularityElement = component.find('.artistCard__info__extra')
      .find({ label: 'Popularity' })
      .render()
      .find('p');
    expect(popularityElement.text()).toBe(POPULARITY_RANGES[0].label);
  });

  it('should render "Cool" when the popularity rate is above 60', () => {
    const popularity = 70;
    component.setProps({ popularity });
    const popularityElement = component.find('.artistCard__info__extra')
      .find({ label: 'Popularity' })
      .render()
      .find('p');

    expect(popularityElement.text()).toBe(POPULARITY_RANGES[1].label);
  });

  it('should render "Regular" when the popularity rate is above 30', () => {
    const popularity = 45;
    component.setProps({ popularity });
    const popularityElement = component.find('.artistCard__info__extra')
      .find({ label: 'Popularity' })
      .render()
      .find('p');

    expect(popularityElement.text()).toBe(POPULARITY_RANGES[2].label);
  });

  it('should render "Underground" when the popularity is under 30', () => {
    const popularity = 15;
    component.setProps({ popularity });
    const popularityElement = component.find('.artistCard__info__extra')
      .find({ label: 'Popularity' })
      .render()
      .find('p');

    expect(popularityElement.text()).toBe(POPULARITY_RANGES[3].label);
  });
});
