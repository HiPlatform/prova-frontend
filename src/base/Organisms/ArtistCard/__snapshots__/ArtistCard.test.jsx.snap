// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCard should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistCard
    details={[Function]}
    genres={
        Array [
            "foo",
            "bar",
            "fuzz",
          ]
    }
    href="https://some.redirect/url"
    images={
        Array [
            Object {
              "url": "https://some.image/url",
            },
          ]
    }
    name="John Due"
    popularity={90}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Image
          alt="album"
          url="https://some.image/url"
/>,
        <div
          className="artistCard__info"
>
          <div
                    className="artistCard__info__title"
          >
                    <AttributeField
                              label={null}
                              text="John Due"
                    />
          </div>
          <div
                    className="artistCard__info__extra"
          >
                    <AttributeField
                              label="Genres"
                              text="foo, bar, fuzz"
                    />
                    <AttributeField
                              label="Popularity"
                              text="Hot"
                    />
          </div>
          <div
                    className="artistCard__info__actions"
          >
                    <Button
                              modifier=""
                              onClick={[Function]}
                    >
                              MORE DETAILS
                    </Button>
          </div>
</div>,
      ],
      "className": "artistCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "album",
          "url": "https://some.image/url",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="artistCard__info__title"
>
              <AttributeField
                            label={null}
                            text="John Due"
              />
</div>,
            <div
              className="artistCard__info__extra"
>
              <AttributeField
                            label="Genres"
                            text="foo, bar, fuzz"
              />
              <AttributeField
                            label="Popularity"
                            text="Hot"
              />
</div>,
            <div
              className="artistCard__info__actions"
>
              <Button
                            modifier=""
                            onClick={[Function]}
              >
                            MORE DETAILS
              </Button>
</div>,
          ],
          "className": "artistCard__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <AttributeField
                label={null}
                text="John Due"
/>,
              "className": "artistCard__info__title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": null,
                "text": "John Due",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <AttributeField
                  label="Genres"
                  text="foo, bar, fuzz"
/>,
                <AttributeField
                  label="Popularity"
                  text="Hot"
/>,
              ],
              "className": "artistCard__info__extra",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Genres",
                  "text": "foo, bar, fuzz",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": "Popularity",
                  "text": "Hot",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                modifier=""
                onClick={[Function]}
>
                MORE DETAILS
</Button>,
              "className": "artistCard__info__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "MORE DETAILS",
                "modifier": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "MORE DETAILS",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Image
            alt="album"
            url="https://some.image/url"
/>,
          <div
            className="artistCard__info"
>
            <div
                        className="artistCard__info__title"
            >
                        <AttributeField
                                    label={null}
                                    text="John Due"
                        />
            </div>
            <div
                        className="artistCard__info__extra"
            >
                        <AttributeField
                                    label="Genres"
                                    text="foo, bar, fuzz"
                        />
                        <AttributeField
                                    label="Popularity"
                                    text="Hot"
                        />
            </div>
            <div
                        className="artistCard__info__actions"
            >
                        <Button
                                    modifier=""
                                    onClick={[Function]}
                        >
                                    MORE DETAILS
                        </Button>
            </div>
</div>,
        ],
        "className": "artistCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "album",
            "url": "https://some.image/url",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="artistCard__info__title"
>
                <AttributeField
                                label={null}
                                text="John Due"
                />
</div>,
              <div
                className="artistCard__info__extra"
>
                <AttributeField
                                label="Genres"
                                text="foo, bar, fuzz"
                />
                <AttributeField
                                label="Popularity"
                                text="Hot"
                />
</div>,
              <div
                className="artistCard__info__actions"
>
                <Button
                                modifier=""
                                onClick={[Function]}
                >
                                MORE DETAILS
                </Button>
</div>,
            ],
            "className": "artistCard__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <AttributeField
                  label={null}
                  text="John Due"
/>,
                "className": "artistCard__info__title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "label": null,
                  "text": "John Due",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <AttributeField
                    label="Genres"
                    text="foo, bar, fuzz"
/>,
                  <AttributeField
                    label="Popularity"
                    text="Hot"
/>,
                ],
                "className": "artistCard__info__extra",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "label": "Genres",
                    "text": "foo, bar, fuzz",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "label": "Popularity",
                    "text": "Hot",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Button
                  modifier=""
                  onClick={[Function]}
>
                  MORE DETAILS
</Button>,
                "className": "artistCard__info__actions",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "MORE DETAILS",
                  "modifier": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "MORE DETAILS",
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
